compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -419,39 +419,43 @@ void
 OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator * cg, TR::CodeGenPhase * phase)
    {
    TR::Compilation *comp = cg->comp();
-   // TODO (GuardedStorage)
-#if defined(OMR_GC_CONCURRENT_SCAVENGER)
-   traceMsg(comp, "GuardedStorage: in performSetupForInstructionSelectionPhase\n");
 
-   auto mapAllocator = getTypedAllocator<std::pair<TR::TreeTop*, TR::TreeTop*> >(comp->allocator());
-
-   std::map<TR::TreeTop*, TR::TreeTop*, std::less<TR::TreeTop*>, TR::typed_allocator<std::pair<TR::TreeTop*, TR::TreeTop*>, TR::Allocator> >
-      currentTreeTopToappendTreeTop(std::less<TR::TreeTop*> (), mapAllocator);
-
-   for (TR::PreorderNodeIterator iter(comp->getStartTree(), comp); iter != NULL; ++iter)
+   if (cg->isConcurrentScavengeEnabled())
```

Comment:
I think this code needs to make a distinction between whether concurrent scavenge
is SUPPORTED by a particular code generator versus whether it is ENABLED at all.

The question that you want to be asking at this point in the code is whether
concurrent scavenge is both ENABLED and SUPPORTED, unless I am totally misreading
what you're trying to do here.  Looking into how this was actually implemented in
`OMR::Z::CodeGenerator::isConcurrentScavengeEnabled`, this query only returns
true if a particular Z processor feature is available, as opposed to whether
concurrent scavenge is actually enabled by the GC in this VM.

Also, as written, this code path will always be taken on Z regardless of the GC
policy in place.  Was that your intent?